rust/lance/src/dataset.rs

```diff
@@ -445,6 +447,9 @@ impl Dataset {
             )
         };
 
+        deblob_schema(&mut schema);
```

Comment:
In an ideal world, there would be a separate Rust type for schemas that are blobed / deblobed. There's probably a better name for it, like Logical/Physical schema. But an actual separate type means it could go in the signature of functions.